components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/client/IdentityProviderMgtServiceClient.java

```diff
@@ -113,22 +122,116 @@ public void updateResidentIdP(IdentityProvider identityProvider) throws Exceptio
+    public IdentityProvider[] getIdPs() throws IdentityException {
+
         try {
-            IdentityProvider[] identityProviders = idPMgtStub.getAllIdPs();
-            if (identityProviders != null && identityProviders.length > 0) {
-                return Arrays.asList(identityProviders);
-            } else {
-                return new ArrayList<IdentityProvider>();
-            }
-        } catch (Exception e) {
+            return idPMgtStub.getAllIdPs();
+        } catch (RemoteException | IdentityProviderMgtServiceIdentityProviderManagementExceptionException e) {
             log.error("Error in retrieving the list of Identity Providers for a given tenant", e);
-            throw new Exception("Error occurred while retrieving list of Identity Providers");
+            throw new IdentityException("Error occurred while retrieving list of Identity Providers", e);
+        }
+    }
+
+    /**
+     * Get all list of Idp information with pagination.
+     *
+     * @param pageNumber pageNumber need to filter.
+     * @return list of paginated idp array.
+     * @throws Exception Error when getting list of Identity Providers.
+     */
+    public IdentityProvider[] getAllPaginatedIdpInfo(int pageNumber) throws IdentityException {
+
+        try {
+            return idPMgtStub.getAllPaginatedIdpInfo(pageNumber);
+        } catch (RemoteException | IdentityProviderMgtServiceIdentityProviderManagementExceptionException e) {
+            throw new IdentityException("Error in retrieving the list of Identity Providers for a given tenant", e);
+        }
+    }
+
+    /**
+     * Get all list of Idp information for a matching filter with pagination.
+     *
+     * @param pageNumber pageNumber need to filter.
+     * @param filter     Application name filter.
+     * @return list of paginated idp array.
+     * @throws Exception Error when getting list of Identity Providers for a matching filter with pagination.
+     */
+    public IdentityProvider[] getPaginatedIdpInfo(int pageNumber, String filter) throws IdentityException {
+
+        try {
+            return idPMgtStub.getPaginatedIdpInfo(filter, pageNumber);
+        } catch (RemoteException | IdentityProviderMgtServiceIdentityProviderManagementExceptionException e) {
+            throw new IdentityException(
+                    "Error in retrieving the list of filtered Identity Providers for a given tenant", e);
+        }
+    }
+
+    /**
+     * Get count of all Identity providers.
+     *
+     * @return total number of Idps.
+     * @throws IdentityException Error when getting count of total Identity Providers.
+     */
+    public int getIdpCount() throws IdentityException {
+
+        try {
+            return idPMgtStub.getAllIdpCount();
+        } catch (RemoteException | IdentityProviderMgtServiceIdentityProviderManagementExceptionException e) {
+            throw new IdentityException("Error in retrieving the count of Identity Providers for a given tenant", e);
```

Comment:
as this method is directly called from UI, let's avoid adding 'e' to the error since it will show in the error message in UI
log the error before throwing